cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -244,21 +245,20 @@ func createDNSService(dnsService *v1.Service, serviceBytes []byte, client client
 	return nil
 }
 
-// getDNSIP fetches the kubernetes service's ClusterIP and appends a "0" to it in order to get the DNS IP
-func getDNSIP(client clientset.Interface) (net.IP, error) {
-	k8ssvc, err := client.CoreV1().Services(metav1.NamespaceDefault).Get("kubernetes", metav1.GetOptions{})
+// getDNSIP returns a dnsIP, which is 10th IP in svcSubnet CIDR range
+func getDNSIP(cfg *kubeadmapi.MasterConfiguration) (net.IP, error) {
```

Comment:
while you're here, please add an unit test for this function

Reply:
and a nit: I'd prefer if you only passed `serviceSubnet string` here, not the full `cfg` object